src/sdam/topology.ts

```diff
@@ -814,6 +816,21 @@ function updateServers(topology: Topology, incomingServerDescription?: ServerDes
     const server = topology.s.servers.get(incomingServerDescription.address);
     if (server) {
       server.s.description = incomingServerDescription;
+      if (
+        incomingServerDescription.error instanceof MongoError &&
+        incomingServerDescription.error.hasErrorLabel(MongoErrorLabel.ResetPool)
+      ) {
+        server.s.pool.clear();
+      } else {
+        const newTopologyType = topology.s.description.type;
```

Comment:
If `incomingServerDescription.error` is `instanceof Error` we will proceed here, should we check that error is nullish?

Reply:
probably not, but it wasn't explicitly called out in the spec, we can add the extra check to avoid the extra work